### PR TITLE
V9: Backport #13029 (thread context.Context in tctl)

### PR DESCRIPTION
**Pull request by strideynet**
Backports #13029 

Enterprise PR: https://github.com/gravitational/teleport.e/pull/460

### CODE REVIEW

**Review by ravicious — APPROVED** [review]

**Review by espadolini — APPROVED** [review]
Only merge this after updating the `e` ref to point to the merged commit in `teleport.e:branch/v9`

**Pull request merged**
